packages/xo-web/src/xo-app/dashboard/health/index.js

```diff
@@ -681,6 +691,21 @@ export default class Health extends Component {
 
   _getOrphanVmSnapshots = createFilter(() => this.props.orphanVmSnapshots, this._getPoolPredicate)
 
+  _getOldSnapshots = createSelector(
+    () => this.props.snapshots,
+    () => this.props.schedules,
+    (snapshots, schedules) =>
+      Object.values(snapshots).filter(snapshot => {
+        const isOld = snapshot.snapshot_time * 1000 < Date.now() - THIRTY_DAYS
+        if (!isOld) return false
+
+        const scheduleId = snapshot.other?.['xo:backup:schedule']
+        const schedule = scheduleId !== undefined ? schedules?.[scheduleId] : undefined
+
+        return !schedule?.enabled
+      })
+  )
```

Comment:
You can inline and take `Date.now()` out of the loop to avoid evaluating it many times in case there are many snapshots.
```suggestion
  _getOldSnapshots = createSelector(
    () => this.props.snapshots,
    () => this.props.schedules,
    (snapshots, schedules) => {
      const thresholdDate = Date.now() - THIRTY_DAYS
      return Object.values(snapshots).filter(snapshot => {
        if (snapshot.snapshot_time * 1000 > thresholdDate) {
          return false
        }

        const scheduleId = snapshot.other?.['xo:backup:schedule']
        const schedule = scheduleId !== undefined ? schedules?.[scheduleId] : undefined

        return !schedule?.enabled
      })
    }
  )
```